app/src/main/java/com/mapbox/maps/testapp/utils/NavigationSimulator.kt

```diff
@@ -256,6 +330,14 @@ class NavigationSimulator(
     private const val GEOJSON_SOURCE_ID = "source_id"
     private const val ROUTE_LINE_LAYER_ID = "route_line_layer_id"
   }
+
+  override fun onIndicatorPositionChanged(point: Point) {
+    val progress = TurfMeasurement.length(
+      TurfMisc.lineSlice(routeStartPoint, point, routePoints),
+      TurfConstants.UNIT_CENTIMETERS
+    ) / totalRouteLength
+    routeLayer.lineTrimOffset(listOf(0.0, progress))
```

Comment:
calculation is a lot more straightforward vs what I expected! nice

Reply:
However it also suffered from precision problem, after some amount of time, the vanishing line moves ahead of the puck...